diff -u pyewf/pyewf_file_entry.c pyewf/pyewf_file_entry.c
--- pyewf/pyewf_file_entry.c	2017-11-05 10:37:45.000000000 +0000
+++ pyewf/pyewf_file_entry.c	2018-09-25 13:59:08.613452224 +0000
@@ -177,6 +177,13 @@
 	  "\n"
 	  "Retrieves the SHA1 hash of the file entry data." },
 
+	{ "get_file_type",
+	  (PyCFunction) pyewf_file_entry_get_file_type,
+	  METH_NOARGS,
+	  "get_file_type() -> Integer\n"
+	  "\n"
+	  "Returns the file type of the file entry." },
+
 	/* Functions to access the sub file entries */
 
 	{ "get_number_of_sub_file_entries",
@@ -259,6 +266,13 @@
 	  "The sub file entries",
 	  NULL },
 
+	{ "file_type",
+		(getter) pyewf_file_entry_get_file_type,
+		(setter) 0,
+		"The file type of the file entry.",
+		NULL },
+
+
 	/* Sentinel */
 	{ NULL, NULL, NULL, NULL, NULL }
 };
@@ -1871,3 +1885,53 @@
 	return( file_entries_object );
 }
 
+/* Retrieves the file type
+ * Returns a Python object if successful or NULL on error
+ */
+PyObject *pyewf_file_entry_get_file_type(
+           pyewf_file_entry_t *pyewf_file_entry,
+           PyObject *arguments PYEWF_ATTRIBUTE_UNUSED )
+{
+	libcerror_error_t *error = NULL;
+	PyObject *integer_object = NULL;
+	static char *function    = "pyewf_file_entry_get_file_type";
+	uint8_t type             = 0;
+	int result               = 0;
+
+	PYEWF_UNREFERENCED_PARAMETER( arguments )
+
+	if( pyewf_file_entry == NULL )
+	{
+		PyErr_Format(
+		 PyExc_TypeError,
+		 "%s: invalid file entry.",
+		 function );
+
+		return( NULL );
+	}
+	Py_BEGIN_ALLOW_THREADS
+
+	result = libewf_file_entry_get_type(
+	          pyewf_file_entry->file_entry,
+	          &type,
+	          &error );
+
+	Py_END_ALLOW_THREADS
+
+	if( result != 1 )
+	{
+		pyewf_error_raise(
+		 error,
+		 PyExc_IOError,
+		 "%s: unable to retrieve file type.",
+		 function );
+
+		libcerror_error_free(
+		 &error );
+
+		return( NULL );
+	}
+	integer_object = PyLong_FromLong((long) type);
+
+	return( integer_object );
+}
diff -u -x '*.lo' -x '*.o' -x '*.lo' -x '*.la' -x .deps -x Makefile -x .libs libewf-20171104/pyewf/pyewf_file_entry.h libewf/pyewf/pyewf_file_entry.h
--- pyewf/pyewf_file_entry.h	2017-11-05 10:37:45.000000000 +0000
+++ pyewf/pyewf_file_entry.h	2018-09-24 21:11:38.823495568 +0000
@@ -147,9 +147,12 @@
            pyewf_file_entry_t *pyewf_file_entry,
            PyObject *arguments );
 
+PyObject *pyewf_file_entry_get_file_type(
+          pyewf_file_entry_t *pyewf_file_entry,
+          PyObject *arguments );
+
 #if defined( __cplusplus )
 }
 #endif
 
 #endif /* !defined( _PYEWF_FILE_ENTRY_H ) */
-